File: Python/DP - 1D/746. Min Cost Climbing Stairs.py
""" 746. Min Cost Climbing Stairs
Difficulty: Easy
Submission Link: https://leetcode.com/problems/min-cost-climbing-stairs/submissions/1377985526/
 """

class Solution(object):
    def minCostClimbingStairs(self, cost):
        for i in range(2, len(cost)):
            cost[i] += min(cost[i-1], cost[i-2])
        return min(cost[-1], cost[-2])

File: C#/DP - 1D/746. Min Cost Climbing Stairs.cs
/* 746. Min Cost Climbing Stairs
Difficulty: Easy
Submission Link: https://leetcode.com/problems/min-cost-climbing-stairs/submissions/1377986432/
 */

 public class Solution {
    public int MinCostClimbingStairs(int[] cost) {
        int n = cost.Length;
        int[] dp = new int[n + 1];
        
        for (int i = 2; i <= n; i++) {
            dp[i] = Math.Min(dp[i - 1] + cost[i - 1], dp[i - 2] + cost[i - 2]);
        }
        
        return dp[n];
    }
}